bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/LinkyHandlerFactory.java

```diff
@@ -50,24 +62,52 @@
  * The {@link LinkyHandlerFactory} is responsible for creating things handlers.
  *
  * @author Gaël L'hopital - Initial contribution
+ * @author Laurent Arnal - Rewrite addon to use official dataconect API
  */
 @NonNullByDefault
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.linky")
-public class LinkyHandlerFactory extends BaseThingHandlerFactory {
+public class LinkyHandlerFactory extends BaseThingHandlerFactory implements LinkyAccountHandler {
     private static final DateTimeFormatter LINKY_FORMATTER = DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSSX");
+    private static final DateTimeFormatter LINKY_LOCALDATE_FORMATTER = DateTimeFormatter.ofPattern("uuuu-MM-dd");
+    private static final DateTimeFormatter LINKY_LOCALDATETIME_FORMATTER = DateTimeFormatter
+            .ofPattern("uuuu-MM-dd HH:mm:ss");
     private static final int REQUEST_BUFFER_SIZE = 8000;
 
+    private final String clientId = "e551937c-5250-48bc-b4a6-2323af68db92";
+    private final String clientSecret = "";
+
     private final Logger logger = LoggerFactory.getLogger(LinkyHandlerFactory.class);
-    private final Gson gson = new GsonBuilder().registerTypeAdapter(ZonedDateTime.class,
-            (JsonDeserializer<ZonedDateTime>) (json, type, jsonDeserializationContext) -> ZonedDateTime
-                    .parse(json.getAsJsonPrimitive().getAsString(), LINKY_FORMATTER))
+    private final Gson gson = new GsonBuilder()
+            .registerTypeAdapter(ZonedDateTime.class,
+                    (JsonDeserializer<ZonedDateTime>) (json, type, jsonDeserializationContext) -> ZonedDateTime
+                            .parse(json.getAsJsonPrimitive().getAsString(), LINKY_FORMATTER))
+            .registerTypeAdapter(LocalDate.class,
+                    (JsonDeserializer<LocalDate>) (json, type, jsonDeserializationContext) -> LocalDate
+                            .parse(json.getAsJsonPrimitive().getAsString(), LINKY_LOCALDATE_FORMATTER))
+            .registerTypeAdapter(LocalDateTime.class,
+                    (JsonDeserializer<LocalDateTime>) (json, type, jsonDeserializationContext) -> {
+                        try {
+                            return LocalDateTime.parse(json.getAsJsonPrimitive().getAsString(),
+                                    LINKY_LOCALDATETIME_FORMATTER);
+                        } catch (Exception ex) {
+                            return LocalDate.parse(json.getAsJsonPrimitive().getAsString(), LINKY_LOCALDATE_FORMATTER)
+                                    .atStartOfDay();
+                        }
+                    })
+
             .create();
+
     private final LocaleProvider localeProvider;
     private final HttpClient httpClient;
+    private final LinkyAuthService authService;
+    private final boolean oAuthSupport = false;
+    private @Nullable OAuthClientService oAuthService;
```

Comment:
Is this really `@Nullable` ?

Reply:
No, not need.
Il modify this in new version.